pkg/render/node.go

```diff
@@ -161,56 +165,67 @@ func (c *nodeComponent) SupportedOSType() rmeta.OSType {
 }
 
 func (c *nodeComponent) Objects() ([]client.Object, []client.Object) {
```

Comment:
I may be confused by these changes because I don't understand how this works in k8s, but do you actually need to / should you explicitly delete anything here if we're terminating?

If the installation resource has been deleted, everything that doesn't have a finalizer will be automatically removed, and everything with a finalizer will be kept until the finalizers are removed.

I think you might introduce a weird race by returning objects to create as well. If kube controllers is deleted and this reconciles then the finalizers will be removed _and_ we might try to update / create these resources, which could lead to some funky behaviour.

I would think the best thing to do here is to return nothing if we're terminating...

Reply:
If we're terminating then we need to re-render (update) the resources that remained due to the finalizers on them previously so they can be cleaned up. Specifically because prior to "terminating": the SA, ClusterRoleBinding, and ClusterRole had a finalizer to prevent their deletion.

Reply:
Oh right, I was (incorrectly) thinking the finalizer was removed separate to this.